A32NX/html_ui/Pages/A32NX_Core/A32NX_FWC.js

```diff
@@ -56,6 +56,16 @@ class A32NX_FWC {
 
         // ESDL 1. 0.320
         this.memoLdgInhibit_conf01 = new NXLogic_ConfirmNode(3, true); // CONF 01
+
+        // master warning & caution buttons
+        this.warningPressed = false;
+        this.cautionPressed = false;
+
+        // altitude warning
+        this.previousTargetAltitude = NaN;
+        this._wasBellowThreshold = false;
+        this._wasAboveThreshold = false;
+        this._wasInRange = false;
```

Comment:
I was wondering whether saving previous delta or use of a state machine would reduce number of these variables.

Then it could be something like:
* `if prevDelta > 750 && currDelta <= 750` -> `ALT_DEVIATION_SHORT` (1.5s)
* `if prevDelta <= 250 && currDelta > 250` -> `ALT_DEVIATION` (continuous)
* `if 750 >= prevDelta > 250 && currDelta > 750` -> `ALT_DEVIATION` (continuous)

Of course there would have to be conditions incorporated in case the warring is inhibited.
The continuous sound/boolean would get precedence over short one.

With this I see one problem of sounding again when the button is pushed between 750ft and 250ft when increasing deviation from selected altitude after passing deviation over 750 ft. I'm not sure if it is a problem or a feature.

If the continuous cchord should stop when reaching 750ft, then it could reset on the same condition as for the short (1.5) cchord.

(pardon my pseudo code)

Reply:
thanks for your feedback. i try to reduce simvars and change some code on that.
i just using pfd code(copy & paste) but try to re-build that :)

Reply:
actually this code is using on PFD so i think it work well.

so i try to reduce simvar only :)
i think this will working well but if there is a error on qa test, then fix that!